internal/handlers/common/filter.go

```diff
@@ -248,7 +248,7 @@ func filterFieldExpr(doc *types.Document, filterKey string, expr *types.Document
 				return false, nil
 			case *types.Array:
 				if fieldValue, ok := fieldValue.(*types.Array); ok {
-					return matchArrays(exprValue, fieldValue), nil
+					return types.CompareArrays(exprValue, fieldValue) == types.Equal, nil
 				}
 				return false, nil
```

Comment:
Quick thought: can we use a regular `types.Compare` there instead of both type assertion and a separate function? Can we un-export `CompareArrays`?